snuba/query/logical.py

```diff
@@ -81,3 +79,73 @@ def _transform_expressions_impl(
 
     def _transform_impl(self, visitor: ExpressionVisitor[Expression]) -> None:
         pass
+
+
+class StorageQuery(ProcessableQuery[Storage]):
```

Comment:
TODO: I think this type can be generalized

Reply:
Good question. The main thing is that the `Table` datasaoure does not resemble the entity in any real way and it's only by the time we get to the `StorageProcessing` stage that we can process the clickhouse query. 

Ideally what I would do is remove `Table` as a datasource and probably remove `ClickhouseQuery` later.

Reply:
Why does it matter that Table doesn't resemble Entity?